PROJECT_MCPU/MCPU/DOC/GENERATED/man/man3/ConfirmationWindow.3
.TH "ConfirmationWindow" 3 "MCPU" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfirmationWindow
.SH SYNOPSIS
.br
.PP
.PP
\fR#include <ConfirmationWindow\&.h>\fP
.PP
Inherits System::Windows::Forms::Form\&.
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum class \fBInfoType\fP { \fBINF_WIN\fP = 0, \fBWRN_WIN\fP, \fBERR_WIN\fP }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "delegate void \fBdelegate_button_callback\fP (void)"
.br
.ti -1c
.RI "void \fBopen\fP (void)"
.br
.ti -1c
.RI "void \fBclose\fP (void)"
.br
.ti -1c
.RI "void \fBformInitialization\fP (\fBInfoType\fP wtype, System::String^ info_title, System::String^ info_content)"
.br
.ti -1c
.RI "void \fBsetContentBackground\fP (Image^ img)"
.br
.ti -1c
.RI "\fBConfirmationWindow\fP (Form^ pp, \fBInfoType\fP wtype, System::String^ info_title, System::String^ info_content)"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "HWND \fBwindow\fP"
.br
.ti -1c
.RI "bool \fBopen_status\fP"
.br
.ti -1c
.RI "\fBInfoType\fP \fBwType\fP"
.br
.ti -1c
.RI "Form ^ \fBparent\fP"
.br
.in -1c
.SS "Protected Member Functions"

.in +1c
.ti -1c
.RI "\fB~ConfirmationWindow\fP ()"
.br
.RI "Clean up any resources being used\&. "
.in -1c
.SS "Events"

.in +1c
.ti -1c
.RI "\fBdelegate_button_callback\fP^ \fBbutton_canc_event\fP"
.br
.ti -1c
.RI "\fBdelegate_button_callback\fP^ \fBbutton_ok_event\fP"
.br
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "System::Void \fBInitializeComponent\fP ()"
.br
.ti -1c
.RI "System::Void \fBCancButton_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBOkButton_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBInfoTitleText\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBInfoContentText\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBmainPanel\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBCancButton\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBOkButton\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBIconPanel\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBContentPanel\fP"
.br
.ti -1c
.RI "System::ComponentModel::Container ^ \fBcomponents\fP"
.br
.RI "Required designer variable\&. "
.in -1c
.SH "Member Enumeration Documentation"
.PP 
.SS "enum class \fBConfirmationWindow::InfoType\fP\fR [strong]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\f(BIINF_WIN \fP(0)
.TP
\f(BIWRN_WIN \fP
.TP
\f(BIERR_WIN \fP
.SH "Constructor & Destructor Documentation"
.PP 
.SS "ConfirmationWindow::ConfirmationWindow (Form^ pp, \fBInfoType\fP wtype, System::String^ info_title, System::String^ info_content)\fR [inline]\fP"

.SS "ConfirmationWindow::~ConfirmationWindow ()\fR [inline]\fP, \fR [protected]\fP"

.PP
Clean up any resources being used\&. 
.SH "Member Function Documentation"
.PP 
.SS "System::Void ConfirmationWindow::CancButton_Click (System::Object^ sender, System::EventArgs^ e)\fR [inline]\fP, \fR [private]\fP"

.SS "void ConfirmationWindow::close (void )"

.SS "delegate void ConfirmationWindow::delegate_button_callback (void )"

.SS "void ConfirmationWindow::formInitialization (\fBInfoType\fP wtype, System::String^ info_title, System::String^ info_content)"

.SS "System::Void ConfirmationWindow::InitializeComponent ()\fR [inline]\fP, \fR [private]\fP"

.SS "System::Void ConfirmationWindow::OkButton_Click (System::Object^ sender, System::EventArgs^ e)\fR [inline]\fP, \fR [private]\fP"

.SS "void ConfirmationWindow::open (void )"

.SS "void ConfirmationWindow::setContentBackground (Image^ img)"

.SH "Member Data Documentation"
.PP 
.SS "System::Windows::Forms::Panel ^ ConfirmationWindow::CancButton\fR [private]\fP"

.SS "System::ComponentModel::Container ^ ConfirmationWindow::components\fR [private]\fP"

.PP
Required designer variable\&. 
.SS "System::Windows::Forms::Panel ^ ConfirmationWindow::ContentPanel\fR [private]\fP"

.SS "System::Windows::Forms::Panel ^ ConfirmationWindow::IconPanel\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ConfirmationWindow::InfoContentText\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ConfirmationWindow::InfoTitleText\fR [private]\fP"

.SS "System::Windows::Forms::Panel ^ ConfirmationWindow::mainPanel\fR [private]\fP"

.SS "System::Windows::Forms::Panel ^ ConfirmationWindow::OkButton\fR [private]\fP"

.SS "bool ConfirmationWindow::open_status"

.SS "Form ^ ConfirmationWindow::parent"

.SS "HWND ConfirmationWindow::window"

.SS "\fBInfoType\fP ConfirmationWindow::wType"

.SH "Event Documentation"
.PP 
.SS "\fBdelegate_button_callback\fP^ ConfirmationWindow::button_canc_event"

.SS "\fBdelegate_button_callback\fP^ ConfirmationWindow::button_ok_event"


.SH "Author"
.PP 
Generated automatically by Doxygen for MCPU from the source code\&.


PROJECT_MCPU/MCPU/MCPU_MASTER/WINDOWS/ConfirmationWindow.cpp
#include "ConfirmationWindow.h"
#include "../gantry_global_status.h"
#include "Log.h"


#define ERR_ICON_IMAGE Image::FromFile(Gantry::applicationResourcePath + "Icons\\error_160x145.PNG")
#define INFO_ICON_IMAGE Image::FromFile(Gantry::applicationResourcePath + "Icons\\info_64x64.PNG")
#define WRN_ICON_IMAGE Image::FromFile(Gantry::applicationResourcePath + "Icons\\info_64x64.PNG")
#define CANC_IMAGE Image::FromFile(Gantry::applicationResourcePath + "Icons\\CANC_ICON.PNG")
#define OK_IMAGE Image::FromFile(Gantry::applicationResourcePath + "Icons\\OK_ICON.PNG")
#define FORM_BACKGROUND Image::FromFile(Gantry::applicationResourcePath + "Icons\\DIALOG_BACKGROUND.PNG")

void ConfirmationWindow::formInitialization(InfoType wtype, System::String^ info_title, System::String^ info_content) {

	// Initialize the position of the form
	// The form position has been set to Center with the parent

	// Set the Form trasparent assigning the background to a transparent key
	// The color is just different from the Black!!
	this->BackColor = Color::FromArgb(1,0,0);
	this->TransparencyKey = Color::FromArgb(1, 0, 0);
	mainPanel->BackgroundImage = FORM_BACKGROUND;
	
	ContentPanel->BackColor = Color::Transparent;
	//

	// Sets the Icon
	switch (wtype) {
	case ConfirmationWindow::InfoType::INF_WIN:
		IconPanel->BackgroundImage = INFO_ICON_IMAGE;
		break;
	case ConfirmationWindow::InfoType::WRN_WIN:
		IconPanel->BackgroundImage = WRN_ICON_IMAGE;
		break;
		
	case ConfirmationWindow::InfoType::ERR_WIN:
		IconPanel->BackgroundImage = ERR_ICON_IMAGE;
		break;
	}
	IconPanel->BackColor = Color::Transparent;

	
	// Sets the Info
	InfoTitleText->Text = info_title;
	InfoTitleText->BackColor = Color::Transparent;
	InfoContentText->Text = info_content;
	InfoContentText->BackColor = Color::Transparent;
	
	// Sets the Buttons
	CancButton->BackColor = Color::Transparent;
	CancButton->BackgroundImage = CANC_IMAGE;

	OkButton->BackColor = Color::Transparent;
	OkButton->BackgroundImage = OK_IMAGE;

	this->Hide();
	open_status = false;
}

void ConfirmationWindow::open(void) {
	if (open_status) return;

	try{
		open_status = true;
		this->ShowDialog(parent);
		window = static_cast<HWND>(Handle.ToPointer());
	}
	catch (...) {
		open_status = false;
		LogClass::logInFile("ConfirmationWindow::open() Exception!");
	}
	
}
void ConfirmationWindow::close(void) {
	if (!open_status) return;
	open_status = false;
	this->Hide();
}

void ConfirmationWindow::setContentBackground(Image^ img) {
	ContentPanel->BackgroundImage = img;
}